osgp/protocol-adapter-dlms/osgp-dlms/src/main/java/org/opensmartgridplatform/dlms/objectconfig/DlmsProfileValidator.java

```diff
@@ -247,6 +249,23 @@ private static void allCaptureObjectsShouldExist(
     }
   }
 
+  private static void allSourceObjectsShouldExist(
+      final DlmsProfile dlmsProfile, final List<String> validationErrors) {
+    final String validationError =
+        dlmsProfile.getObjects().stream()
+            .filter(object -> object.getClassId() == InterfaceClass.DATA.id())
+            .map(ObjectConfigService::getSourceObjectDefinitions)
+            .flatMap(Collection::stream)
+            .map(ObjectConfigService::getCosemObjectTypeFromCaptureObjectDefinition)
+            .map(tag -> objectShouldExist(dlmsProfile, tag))
+            .filter(error -> !error.isEmpty())
+            .collect(Collectors.joining(", "));
+
+    if (!validationError.isEmpty()) {
```

Comment:
Deze if wordt niet gedekt in de unit tests.

Reply:
This test checks the validity the the DLMS profile and should test positive compile time. The definition does not change runtime. If validator errors exist they should be fixed designtime